spec/units/tab_completion/dsl/parser_spec.rb

```diff
@@ -154,6 +154,18 @@
       expect(result).to be_a_text_transition
       expect(result.word).to eq('stash')
     end
+
+    it 'parses fallback rules' do
+      result = parse_single_rule(tokens(
+        [:VAR, 'command'], [:MODIFIER, 'fallback'], [:VAR, 'revision'], [:EOS],
+      ))
+
+      expect(result).to be_a_concatenation
+      expect(result.parts.length).to eq(2)
+      expect(result.parts.first).to be_a_fallback_transition
+      expect(result.parts.first.matcher).to be_a_command_matcher
+      expect(result.parts[1]).to be_a_variable_transition
```

Comment:
`result.parts.first`?

Reply:
I've switched to using `[0]` and `[1]` instead of `.first` and `[1]`, so at least it's consistent now.